### PR TITLE
Fixed (grouped)conv2d layout infer bug 

### PR DESCRIPTION
**Pull request by chxin66**
And added a weight as input case to test

Signed-off-by: Chen Xin <jack.chen@verisilicon.com>

### CODE REVIEW

File: src/tim/transform/ops/conv2d_layout_inference.h [hunk 1]
```diff
@@ -40,63 +40,89 @@ class Conv2dLayoutInfer : public OpLayoutInfer {
...
       std::shared_ptr<vx::Tensor> infer_tensor;
-      std::shared_ptr<IPermuteVector> trans_pv;
       if (in->IsConstTensor() &&
           !(in->GetSpec().attr_ & vx::TensorAttribute::INPUT)) {
-            // For bias
-            if (in->GetShape().size() == 1) {
-              infer_tensor = context_->infer_graph_->CreateTensor(
-                  in->GetSpec(), in->GetDataRef());
-              trans_pv = MakeShared(1);
+        // Handle const weight. Some input may have both CONSTANT & INPUT, not handle here.
```
Comment (sunshinemyson) [inline src/tim/transform/ops/conv2d_layout_inference.h:81]: if not handle here, where to handle it?

File: src/tim/transform/ops/conv2d_layout_inference.h [hunk 2]
```diff
@@ -40,63 +40,89 @@ class Conv2dLayoutInfer : public OpLayoutInfer {
...
+        break;
+      default: // Default set to IWHO for compatibility with previous APIs
+        weight_required_pv = std::make_shared<PermuteVector<4>>(kIcWHOc2WHIcOc);
+        break;
     }
-    auto input_tensors = op_->impl()->InputsTensor();
 
+    auto input_tensors = op_->impl()->InputsTensor();
+    trans_pv = required_pv;
     for (const auto& in : input_tensors) {
```
Comment (sunshinemyson) [inline src/tim/transform/ops/conv2d_layout_inference.h:77]: why not handle it one by one without for-loop? If so, the code may more clean than now.

File: src/tim/transform/ops/conv2d_layout_inference.h (hunk 1 above)
Comment (chxin66) [reply src/tim/transform/ops/conv2d_layout_inference.h:81]: this "if" pass only handle const_weight & const_bias, this is to void a input_tensor attr both has  "INPUT" & "CONSTANT". the tenosr has attr "INPUT" will handle in "else"

File: src/tim/transform/ops/conv2d_layout_inference.h (hunk 2 above)
Comment (chxin66) [reply src/tim/transform/ops/conv2d_layout_inference.h]: done

**Review by sunshinemyson — APPROVED** [review]

**Review by sunshinemyson — APPROVED** [review]

**Pull request merged**